core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala

```diff
@@ -81,6 +84,24 @@ class DeltaLog private(
 
   protected def spark = SparkSession.active
 
+  /**
+   * Issue 1144 - Verify the required Spark conf for delta
+   * Throw `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if both
+   * `spark.sql.extensions` and `spark.sql.catalog.spark_catalog` config are missing.
+   * If `DeltaSparkSessionExtension` is alternatively activated using the `.withExtension()` API,
+   * we expect the DeltaSqlParser to be available and try parse the vacuum command.
+   * If not successful, we throw exception.
+   */
+  if(spark.conf.getOption(StaticSQLConf.SPARK_SESSION_EXTENSIONS.key)
+    .orElse(Try(spark.sessionState.sqlParser.parsePlan("vacuum delta_table")).toOption)
+    .isEmpty ||
+    spark.conf.getOption(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key).isEmpty) {
```

Comment:
These two configs must be set via static configs. Could you check them in `spark.sparkContext.getConf` instead?

Reply:
will do. We also have a case where a SparkSession is initialized with `.withExtensions()` API, instead of using the `.config("spark.sql.extensions", "")` api in which case `spark.sparkContext.getConf` won't be able to find `spark.sql.extensions` config. This is why I also had to try parsing a delta SQL command to see if the Delta SQL parser is available. We still want to do this right?

Reply:
Apparently, `DeltaSQLCommandTest` sets the catalog config using `session.conf.set()` api as shown below:

```
 session.conf.set(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key, classOf[DeltaCatalog].getName)
```
if we use `spark.sparkContext.getConf()` api to check the config, then test classes that extend `DeltaSQLCommandTest` will fail